pkg/mapper/mapper_cache.go

```diff
@@ -56,155 +53,36 @@ func NewCacheMetrics(reg prometheus.Registerer) *CacheMetrics {
 	return &m
 }
 
-type cacheOptions struct {
-	cacheType string
-}
-
-type CacheOption func(*cacheOptions)
-
-func WithCacheType(cacheType string) CacheOption {
-	return func(o *cacheOptions) {
-		o.cacheType = cacheType
-	}
-}
-
 type MetricMapperCacheResult struct {
 	Mapping *MetricMapping
 	Matched bool
 	Labels  prometheus.Labels
 }
 
+// MetricMapperCache must be thread-safe and should be instrumented with CacheMetrics
 type MetricMapperCache interface {
-	Get(metricString string, metricType MetricType) (*MetricMapperCacheResult, bool)
-	AddMatch(metricString string, metricType MetricType, mapping *MetricMapping, labels prometheus.Labels)
-	AddMiss(metricString string, metricType MetricType)
+	// Get a cached result
+	Get(metricKey string) (interface{}, bool)
```

Comment:
What is the benefit of this compared to keeping the metricString and metricType separate?

Reply:
In my mind, it made more sense to build the key from the two parts and pass it into the cache already formatted. This way it makes the cache interface more generic (key-value, vs key1 key2 - value), and also keeps people from having to duplicate the key-building code in each cache package.